packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -153,7 +153,24 @@ export class SelectedNetworkController extends BaseController<
 
     this.messagingSystem.subscribe(
       'NetworkController:stateChange',
-      ({ selectedNetworkClientId }) => {
+      ({ selectedNetworkClientId }, patches) => {
+        patches.forEach(({ op, path }) => {
```

Comment:
Interesting idea to use patches for this. Might be more performant this way? I worry about the potential for a network configuration to be replaced by a different operation though, e.g. move or replace.

At the moment I think remove is the only possibility, but later with data syncing, I am not so sure.

Comparing each selected network client ID with each existing network might be safer.

Reply:
We could ensure this only runs for config changes by using a selector as well, e.g.

```
 this.messagingSystem.subscribe(
      'NetworkController:stateChange',
      (networkConfigurations) => {
        ...
      },
      (state) => state.networkConfigurations
```